Source/NSubstitute/Core/ICallResultsCache.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace NSubstitute.Core
+{
+    public interface ICallResultsCache
```

Comment:
Well, I see it's purpose - to be a standalone registry to track "known" call results. It has both `Add` and `Get` API. To be fair, it isn't just a pure wrapper - [it has some logic](https://github.com/Zvirja/NSubstitute/blob/ea566ae65422ef768f02a7b2d5f6215d6b31a39e/Source/NSubstitute/Core/CallResultsCache.cs#L14) which combines two APIs. 

But given that we can easily substitute `ICallResultsCache` by (`ICallResults`, `ICallSpecificationFactory`) pair - I wouldn't persist too much and will agree to remove it if you feel it's better for further support and such instances just add overhead.